### PR TITLE
Abstract linux DBus and GTK storage providers, use async initialization

### LINKED ISSUE #8537: DBus file dialogs crash the application if
```
Unhandled exception. Tmds.DBus.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files
   at Tmds.DBus.DBusConnection.CallMethodAsync(Message msg, Boolean checkConnected, Boolean checkReplyType)
   at Tmds.DBus.Connection.CallMethodAsync(Message message)
   at Tmds.DBus.CodeGen.DBusObjectProxy.SendMethodReturnReaderAsync(String iface, String member, Nullable`1 inSignature, MessageWriter writer)
   at Tmds.DBus.CodeGen.DBusObjectProxy.CallNonVoidMethodAsync[T](String iface, String member, Nullable`1 inSignature, MessageWriter writer, ReadMethodDelegate`1 readValue)
   at Avalonia.FreeDesktop.DBusSystemDialog.OpenFilePickerAsync(FilePickerOpenOptions options) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.FreeDesktop/DBusSystemDialog.cs:line 69
   at Avalonia.Controls.Platform.SystemDialogImpl.ShowFileDialogAsync(FileDialog dialog, Window parent) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.Controls/Platform/Dialogs/SystemDialogImpl.cs:line 27
   at ControlCatalog.Pages.DialogsPage.<>c__DisplayClass0_0.<<-ctor>b__2>d.MoveNext() in /home/kekekeks/Projects/AvaloniaMaster/samples/ControlCatalog/Pages/DialogsPage.xaml.cs:line 71
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at Avalonia.Threading.JobRunner.JobWithArg.Avalonia.Threading.JobRunner.IJob.Run() in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.Base/Threading/JobRunner.cs:line 241
   at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.Base/Threading/JobRunner.cs:line 38
   at Avalonia.Threading.Dispatcher.RunJobs() in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.Base/Threading/Dispatcher.cs:line 69
   at Avalonia.X11.X11PlatformThreading.HandleX11(CancellationToken cancellationToken) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.X11/X11PlatformThreading.cs:line 199
   at Avalonia.X11.X11PlatformThreading.RunLoop(CancellationToken cancellationToken) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.X11/X11PlatformThreading.cs:line 245
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.Base/Threading/Dispatcher.cs:line 61
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 120
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode) in /home/kekekeks/Projects/AvaloniaMaster/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 209
   at ControlCatalog.NetCore.Program.Main(String[] args) in /home/kekekeks/Projects/AvaloniaMaster/samples/ControlCatalog.NetCore/Program.cs:line 103
   ```
   
   
   Also for some reason DBus file dialogs are enabled by the option that enables the global menu support.

### PR DESCRIPTION
**Pull request by maxkatz6**
## Fixed issues

Fixes https://github.com/AvaloniaUI/Avalonia/issues/8537

### CODE REVIEW

File: src/Avalonia.FreeDesktop/DBusHelper.cs [hunk 1]
```diff
@@ -24,8 +24,7 @@ public override void Post(SendOrPostCallback d, object? state)
                     if (_ctx is not null)
                         _ctx?.Post(d, state);
                     else
-                        lock (_lock)
```
Comment (maxkatz6) [inline src/Avalonia.FreeDesktop/DBusHelper.cs:27]: What was point of double lock here on the same thread?

File: Avalonia.sln [hunk 2]
```diff
@@ -559,6 +559,7 @@ Global
 		{2B390431-288C-435C-BB6B-A374033BD8D1} = {4ED8B739-6F4E-4CD4-B993-545E6B5CE637}
 		{EABE2161-989B-42BF-BD8D-1E34B20C21F1} = {C5A00AC3-B34C-4564-9BDD-2DA473EF4D8B}
 		{1BBFAD42-B99E-47E0-B00A-A4BC6B6BB4BB} = {4ED8B739-6F4E-4CD4-B993-545E6B5CE637}
+		{4D36CEC8-53F2-40A5-9A37-79AAE356E2DA} = {86C53C40-57AA-45B8-AD42-FAE0EFDF0F2B}
```
Comment (maxkatz6) [inline Avalonia.sln:562]: Moves "FreeDesktop" project to the "Linux" folder in the IDE.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022245-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.X11/NativeDialogs/LinuxStorageProvider.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Avalonia.FreeDesktop;
+using Avalonia.Platform.Storage;
+
+namespace Avalonia.X11.NativeDialogs;
+
+internal class LinuxStorageProvider : IStorageProvider
```
Comment (kekekeks) [inline src/Avalonia.X11/NativeDialogs/LinuxStorageProvider.cs:9]: Can we have "CompositeStorageProvider" instead? Something that would accept `IEnumerable<Func<Task<IStorageProvider>>` in constructor.

**Review by kekekeks — APPROVED** [review]
Seems to work without xdg-portal now

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022278-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**